content/news/2022-07-29-bevy-0.8/index.md

```diff
@@ -0,0 +1,1131 @@
+    fn extract_resource(source: &CoolColor) -> Self {
+        Self {
+            color: source.color.as_rgba_linear(),
+        }
+    }
+}
+```
+
+
+[`ExtractResource`]: https://docs.rs/bevy/0.8.0/bevy/render/extract_resource/trait.ExtractResource.html
+[`ExtractResourcePlugin`]: https://docs.rs/bevy/0.8.0/bevy/render/extract_resource/struct.ExtractResourcePlugin.html
+[`App`]: https://docs.rs/bevy/0.8.0/bevy/app/struct.App.html
+
+## Query IntoIter
+
+<div class="release-feature-authors">authors: @TheRawMeatball</div>
+
+Bevy ECS [`Queries`][`Query`] now implement the [`IntoIterator`] trait, which provides access to Rust's nicer iteration syntax:
+
+```rust
+// In previous Bevy versions, iter/iter_mut had to be called manually:
+fn system(mut players: Query<&mut Player>) {
+    for player in players.iter() {
+    }
+    
+    for mut player in players.iter_mut() {
+    }
+}
+
+// In Bevy 0.8, you have the option to use this syntax:
+fn system(mut players: Query<&mut Player>) {
+    for player in &players {
+    }
+    
+    for mut player in &mut players {
+    }
+}
+```
+
+[`IntoIterator`]: https://doc.rust-lang.org/std/iter/trait.IntoIterator.html
+[`Query`]: https://docs.rs/bevy/0.8.0/bevy/ecs/system/struct.Query.html
+
+## Query::iter_many
+
+<div class="release-feature-authors">authors: @devil-ira</div>
+
+It is now possible to pass a list of entities to a [`Query`] to iterate over using [`Query::iter_many`], which is significantly more ergonomic than calling `get(entity)` on each individual entity:
+
+```rust
+#[derive(Component)]
+struct BlueTeam {
+    members: Vec<Entity>,
+}
+
+fn system(blue_team: Res<BlueTeam>, players: Query<&Player>) {
+    info!("Blue Team Assemble!");
+    for player in players.iter_many(&blue_team.members) {
+        info!("{}", player.name);
+    }    
+}
+```
+
+There is also a [`Query::many_for_each_mut`], which provides similar functionality for mutable queries, but it uses a closure to ensure there is never aliased mutability if the given list contains duplicates:
+
+```rust
+players.many_for_each_mut(&blue_team.members, |mut player| {
+    player.score += 1;
+});
+```
+
+[`Query::iter_many`]: https://docs.rs/bevy/0.8.0/bevy/ecs/system/struct.Query.html#method.iter_many
+[`Query::many_for_each_mut`]: https://docs.rs/bevy/0.8.0/bevy/ecs/system/struct.Query.html#method.many_for_each_mut
+
+
+## Convert Mutable Queries to Read-only
+
+<div class="release-feature-authors">authors: @harudagondi</div>
+
+Mutable [`Queries`][`Query`] can now be converted to their read-only versions, which makes it easier to build and use abstractions on queries:
+
+```rust
+fn system(mut players: Query<&mut Player>) {
+    for mut player in &mut players {
+        // mutate players here
+    }
+
+    log_players(players.to_readonly());
+}
+
+fn log_players(players: Query<&Players>) {
+    for player in &players {
+        info!("{player:?}");
+    }
+}
+```
+
+## Scripting / Modding Progress: Untyped ECS APIs
```

Comment:
Belongs directly after lifetimed pointers as it builds on them IMO.

Reply:
Yeah I had this section in mind when I mentioned "strategic re-ordering" in the pr description. I think putting it this far down buries a pretty "big" win that will excite a lot of people. Even though it "logically depends" on lifetimed pointers, I almost want to put it before them. (maybe right after the rendering content).

Reply:
Just bumped this up.